providers/redfish/firmware.go

```diff
@@ -248,6 +315,16 @@ func (c *Conn) runRequestWithMultipartPayload(method, url string, payload map[st
 	return c.redfishwrapper.RunRawRequestWithHeaders(method, url, bytes.NewReader(payloadBuffer.Bytes()), payloadWriter.FormDataContentType(), nil)
 }
 
+func (c *Conn) runRequestWithPayload(method, url string, payload io.Reader) (*http.Response, error) {
+	if url == "" {
+		return nil, fmt.Errorf("unable to execute request, no target provided")
+	}
+
+	b, _ := io.ReadAll(payload)
```

Comment:
I'd like to see if we have other options than reading the whole file into memory

Reply:
The `payloadReadSeeker` needs to be of type `io.ReadSeeker` meaning we need to be able to read and seek anywhere in the data, so we're forced to have it all.

On the multipart update, I see `bytes.NewReader(payloadBuffer.Bytes())`.
I'm not sure if the whole update is not in memory at that time.

Reply:
Thanks for pointing this out, I realised that this was a regression and its supposed to be a stream upload https://github.com/bmc-toolbox/bmclib/pull/345